### ISSUE TITLE
linter: `no-restricted-globals` type error in config for spread

### ISSUE THREAD

**Issue by elusiveunit** [oxc-project/oxc#25666]
### What version of Oxlint are you using?

1.78.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```typescript
const exampleGlobals = ['addEventListener', 'blur', 'screen'];

export default defineConfig({
	rules: {
		'no-restricted-globals': [
			'warn',
			...exampleGlobals.map((name) => ({
				name,
				message: `Use globalThis.${name} if intentional.`,
			})),
		],
	},
});
```

### What happened?

The config works (linting `console.log(screen);` errors with `Unexpected use of 'screen'. Use globalThis.screen if intentional.`) but a type error is reported in the config.
```
oxlint.config.ts:7:3 - error TS2322: Type '["warn", ...{ name: string; message: string; }[]]' is not assignable to type 'RuleNoConfig | [AllowWarnDeny, NoRestrictedGlobalsConfigEnum, ...NoRestrictedGlobalsConfigEnum[]] | undefined'.
  Type '["warn", ...{ name: string; message: string; }[]]' is not assignable to type '[AllowWarnDeny, NoRestrictedGlobalsConfigEnum, ...NoRestrictedGlobalsConfigEnum[]]'.
    Source provides no match for required element at position 1 in target.

7   'no-restricted-globals': [
    ~~~~~~~~~~~~~~~~~~~~~~~
```

* Just spreading the strings as `'no-restricted-globals': ['warn', ...exampleGlobals]` produces a similar `Type '["warn", ...string[]]' is not assignable...` error.
* Adding the strings inline as `'no-restricted-globals': ['warn', 'addEventListener', 'blur', 'screen']` works.
* Real example uses https://www.npmjs.com/package/confusing-browser-globals.

Guessing #24598 is related.
